src/Microsoft.VisualStudio.Web.CodeGeneration.Tools/TargetInstaller.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+
+namespace Microsoft.VisualStudio.Web.CodeGeneration.Tools
+{
+    public class TargetInstaller
+    {
+        private ILogger _logger;
+
+        public TargetInstaller(ILogger logger)
+        {
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            _logger = logger;
+        }
+
+
+        // Always overwrite the file as we include the version of the package in here.
```

Comment:
Always overwriting can probably be avoided by parameterizing your targets file. i.e. make the version an MSBuild proeprty that you include when invoking msbuild. `dotnet msbuild /p:SomeVersion=blah`